File: CONTIGIOUS-ARRAY/ca.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int findMaxLength(int[] nums) {
        Map<Integer, Integer> countMap = new HashMap<>();
        int count = 0;
        int maxLen = 0;

        countMap.put(0, -1);

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                count--;
            } else {
                count++;
            }

            if (countMap.containsKey(count)) {
                maxLen = Math.max(maxLen, i - countMap.get(count));
            } else {
                countMap.put(count, i);
            }
        }

        return maxLen;
    }
}

File: CONTIGIOUS-ARRAY/ca.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int, int> count_map;
        int count = 0;
        int maxlen = 0;

        count_map[0] = -1;

        for(int i = 0; i < nums.size(); i++){
            if(nums[i] == 0){
                count --;
            }
            else{
                count ++;
            }
            if (count_map.find(count) != count_map.end()){
                maxlen = max(maxlen, i - count_map[count]);
            }
            else{
                count_map[count] = i;
            }
        }
    return maxlen;
    }
};